isis/src/base/apps/framestitch/framestitch.cpp

```diff
@@ -0,0 +1,220 @@
+#include "framestitch.h"
+
+#include <QString>
+
+#include "Brick.h"
+#include "Buffer.h"
+#include "Cube.h"
+#include "CubeAttribute.h"
+#include "Camera.h"
+#include "Histogram.h"
+#include "IException.h"
+#include "IString.h"
+#include "LineManager.h"
+#include "ProcessByBrick.h"
+#include "PvlGroup.h"
+#include "PvlKeyword.h"
+#include "Statistics.h"
+#include "UserInterface.h"
+
+using namespace std;
+using namespace Isis;
+
+namespace Isis {
+
+  /**
+   * Helper function to count the size of frames in a cube
+   */
+  int computeFrameSize(Cube *cube) {
+    // Use a histogram to accumulate frame counts
+    Histogram frameSizes(0.0, 1024.0, 1024);
+    LineManager cubeLine(*cube);
+    int currentFrameSize = 0;
+    for (int line = 1; line <= cube->lineCount(); line++) {
+      cubeLine.SetLine(line);
+      cube->read(cubeLine);
+      Statistics lineStats;
+      lineStats.AddData(cubeLine.DoubleBuffer(), cubeLine.size());
+      // If the line is all NULL add it to the current frame count
+      if (lineStats.TotalPixels() == lineStats.NullPixels()) {
+        ++currentFrameSize;
+      }
+      // If the line has non-NULL pixels and we have previously counted
+      // some all null lines add the count to our histogram and reset
+      else if (currentFrameSize > 0) {
+        frameSizes.AddData(currentFrameSize);
+        currentFrameSize = 0;
+      }
+    }
+
+    // If the last line is part of a NULL frame handle it now
+    if (currentFrameSize > 0) {
+      frameSizes.AddData(currentFrameSize);
+    }
+
+    // Special case where there are no all NULL lines
+    if (frameSizes.TotalPixels() == 0) {
+      return 0;
+    }
+
+    // Use the median to handle any outliers
+    return frameSizes.Median();
+  }
+
+  /**
+   * Compute the stats for an ISIS cube. This is the programmatic interface to
```

Comment:
This comment also appears to not say what is actually happening. This function does the stitching, rather than the stats.